packages/braid-design-system/lib/components/Box/Box.docs.tsx

```diff
@@ -127,6 +128,39 @@ const docs: ComponentDocs = {
         </>
       ),
     },
+    {
+      label: 'Data attributes',
+      description: (
+        <>
+          <Text>
+            Braid components are very explicit about the properties they accept,
+            which makes providing arbitrary data attributes not possible.
+            Instead, all Braid components accept a <Strong>data</Strong> prop,
+            allowing a single collection of data attributes to be provided.
+          </Text>
+          <Notice>
+            <Text>
+              While Box does support the native HTML syntax, it also supports
+              the <Strong>data</Strong> prop for consistency.
+            </Text>
+          </Notice>
+          <Code playroom={false}>
+            {
+              source(
+                <Box
+                  data={{
+                    testid: 'customIdentifier',
+                  }}
+                >
+                  {/* Becomes: data-testid="customIdentifier" */}
```

Comment:
Feels a bit weird that this comment is inside the component. Would it make more sense one line higher so it's more obvious that it's tied to the `data` prop?

Reply:
Yeah there was some formatting constraints in the code block, but will throw around some alternatives.